website/docs/interactors/4-matchers-assertions.md

```diff
@@ -0,0 +1,228 @@
+
+```js
+Link(not('Google')).has({ href: 'https://twitter.com' });
+```
+
+### some, every
+For when you need to assert against iterables, you will find the `some()` and `every()` matchers very helpful. We will use the [`MultiSelect`](https://github.com/thefrontside/bigtest/blob/v0/packages/interactor/src/definitions/multi-select.ts#L48) interactor for the next example because its `values` filter returns an array based on its options' label:
+
+```js
+.filters({
+  values: (e) => Array.from(e.selectedOptions).map((o) => o.label),
+})
+```
+
+`some()` matcher will return true if the argument matches any one of the array items, and `every()` will return true only if all items match. We will use the select element below to demonstrate how these two matchers work:
+
+```html
+<select multiple>
+  <option selected>Neon Blue</option>
+  <option selected>Neon Green</option>
+</select>
+```
+
+In the following tests, we're going to first assert that _some_ of the select element's options has a value that includes the word "Blue", and then assert that _every_ option starts with "Neon":
+
+```js
+MultiSelect().has({ values: some(including('Blue')) });
+MultiSelect().has({ values: every(matching(/^Neon/)) });
+```
+
+In the two tests above we are passing in the `including()` and `matching()` matchers into `some()` and `every()`. Once again, `and()`, `or()`, `not()`, `some()`, and `every()` can take matchers as its arguments. This means you can chain them together multiple times to cater to your needs.
+
+Though the matchers are already ergonomic, you can make your tests even tidier and easier to read by creating your own matchers.
+
+## Creating matchers
+There are two ways you can write your own matcher: by piggybacking on preexisting matchers or you can create your own from scratch. We will cover both methods in this section.
+
+Let us start by refactoring the last example by creating a matcher called `blueOrGreen`:
+
+```js
+import { including, or } from 'bigtest';
+
+export const blueOrGreen = or(
+  including('Blue'),
+  including('Green')
+);
+```
+
+You can then import and use the new matcher in your tests like so:
+
+```js
+MultiSelect().has({ values: every(blueOrGreen) });
+```
+
+Composing a matcher using other matchers, like we did for `blueOrGreen()`, is convenient because it delegates most of the matcher's logic as well as the error message.
+
+To create your own matcher without the use of any of the preexisting ones, you will need to create a function that returns a `{ match(), format() }` object.
+
+The `match()` function is where you place all of the matcher logic. It takes an argument `actual` which represents the values from the interactors. Here's how the `including()` matcher is implemented:
+
+```js
+export function including(subString) {
+  return {
+    match(actual) {
+      return actual.includes(subString);
+    },
+    format() {
```

Comment:
is `format` only used to format the error message? if so can we name it more explicitly like `formatErrorMessage`? format reads a bit confusing imo because it is also makes me think that I'll use format to check a value complies with a format or something like that

Reply:
What about something like `description()`?

Reply:
Created an issue [here](https://github.com/thefrontside/bigtest/issues/906)